### PR TITLE
[PLAT-415] Migrate to python 3.6

### PR DESCRIPTION
**Pull request by im-abeer**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -4,15 +4,16 @@ ENV DEBIAN_FRONTEND=noninteractive
```
[the remark this answers is not in the document]
Comment (im-abeer) [reply Dockerfile]: Official docker image of `ubuntu:18.04` doesn't come along with python.
We'll have to install python3.6 anyways.

[the remark this answers is not in the document]
Comment (saich) [reply Dockerfile]: One difference would be - It is present in official apt source. We don't have to add a 3rd part repo.

I'm okay to live with `16.04` as well for now.

[the remark this answers is not in the document]
Comment (im-abeer) [reply Dockerfile]: Okay. Keeping `16.04` for now

File: Dockerfile [hunk 2]
```diff
@@ -4,15 +4,16 @@ ENV DEBIAN_FRONTEND=noninteractive
...
     libssl-dev \
     libxrender-dev \
-    fontconfig
+    fontconfig \
+    wget
 
 # Download wkhtmltopdf binary
-RUN apt-get install -y wget && \
-    wget https://github.com/wkhtmltopdf/wkhtmltopdf/releases/download/0.12.4/wkhtmltox-0.12.4_linux-generic-amd64.tar.xz && \
+RUN wget https://github.com/wkhtmltopdf/wkhtmltopdf/releases/download/0.12.4/wkhtmltox-0.12.4_linux-generic-amd64.tar.xz && \
```
[the remark this answers is not in the document]
Comment (im-abeer) [reply Dockerfile]: Okay

File: tests.py [hunk 3]
```diff
@@ -7,9 +7,11 @@ class TestApp(unittest.TestCase):
     url = 'http://localhost:8080/'
 
     def test_json_request(self):
-        data = {
-            'contents': open('testcases/sample.html').read().encode('utf-8'),
-        }
+
+        data = {}
+        with open('testcases/sample.html') as html:
```
[the remark this answers is not in the document]
Comment (im-abeer) [reply tests.py]: Yup. Will update

File: requirements.in [hunk 4]
```diff
@@ -0,0 +1,4 @@
+executor==21.3
+gunicorn==19.9.0
+requests==2.22.0
+werkzeug==0.15.6
```
[the remark this answers is not in the document]
Comment (im-abeer) [reply requirements.in]: Updated

File: Dockerfile [hunk 5]
```diff
@@ -4,34 +4,47 @@ ENV DEBIAN_FRONTEND=noninteractive
...
 RUN apt-get update -y && apt-get install -y \
+    software-properties-common \
     build-essential \
     xorg \
     libssl-dev \
     libxrender-dev \
-    fontconfig
+    libjpeg-turbo8-dev \
+    fontconfig \
+    xfonts-75dpi \
```
Comment (saich) [inline Dockerfile]: New packages?

Comment (im-abeer) [reply Dockerfile]: Yes, `libjpeg-turbo8-dev` and `xfonts-75dpi` are required by version **0.12.5** of `wkhtmltopdf`

**Pull request merged**